### PR TITLE
Create a more structured pyproject.toml object

### PR DESCRIPTION
**Pull request by abn**
This change is a step towards making interactions with pyproject.toml
files more structured and standardised.

Resolves: python-poetry/poetry#2851

- [x] Added **tests** for changed code.

### CODE REVIEW

File: poetry/core/exceptions/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from poetry.core.exceptions.base import PoetryCoreException
+
+
+__all__ = [i.__name__ for i in {PoetryCoreException}]
```
Comment (stephsamson) [inline poetry/core/exceptions/__init__.py:4]: Can we make `i` more descriptive?

File: poetry/core/pyproject/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from poetry.core.pyproject.exceptions import PyProjectException
+from poetry.core.pyproject.tables import BuildSystem
+from poetry.core.pyproject.toml import PyProjectTOML
+from poetry.core.pyproject.toml import PyProjectTOMLFile
+
+
+__all__ = [
+    i.__name__
+    for i in {BuildSystem, PyProjectException, PyProjectTOML, PyProjectTOMLFile}
```
Comment (stephsamson) [inline poetry/core/pyproject/__init__.py:9]: Same comment as above.

File: poetry/core/pyproject/toml.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            )
+
+    def __getattr__(self, item):
+        return getattr(self.__path, item)
+
+    def __str__(self):  # type: () -> str
+        return self.__path.as_posix()
+
+
+class PyProjectTOML:
```
Comment (stephsamson) [inline poetry/core/pyproject/toml.py:42]: 💯

**Comment by abn** [comment]
@stephsamson this should be ready again.

**Review by stephsamson — APPROVED** [review]

**Pull request merged**